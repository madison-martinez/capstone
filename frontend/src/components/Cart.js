import React from 'react';
import { connect } from 'react-redux';
import CartStyles from '../components/styles/CartStyles'
import formatMoney from '../utils/formatMoney';
import CartProduct from './CartProduct';


function Cart({ cartProducts, totalPrice}) {

    return (
        <>
           {!cartProducts.length && (
               <h1>Your cart looks pretty empty.</h1>
           )}
           <CartProduct />
           <h2>Your total is: {formatMoney(totalPrice)}</h2>
        </>
    )
};

const mapStateToProps = state => ({
    cartCount: state.cart.cartCount,
    totalPrice: state.cart.totalPrice,
    cartProducts: state.cart.cartProducts
});

export default connect(mapStateToProps)(Cart)