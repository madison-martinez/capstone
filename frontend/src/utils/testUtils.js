import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { reducers } from '../reducers';

function render(
  ui,
  {
    initialState,
    store = createStore(reducers, initialState ),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

//mock middleware functions
const thunk = ({ dispatch, getState }) => next => action => {
    if (typeof action === 'function') {
      return action(dispatch, getState)
    }
    return next(action)
};

const fakeItem = () => ({
    id: '1',
    price: 20.00,
    title: 'Roses',
    description: 'Roses are Red',
});

export {
    fakeItem,
    thunk
};

export * from '@testing-library/react';
export { render };