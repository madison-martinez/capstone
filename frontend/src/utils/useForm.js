import { useState } from 'react';



export default function useForm(initialState = {}) {
    const [values, setValues] = useState(initialState);

    function handleChange(e) {
        let { value, name, type } = e.target;

        if (type === 'select') {
            value = e.target.value;
        }

        setValues({
            ...values,
            [name]: value
        });
    };
    
    async function handleImageUpload(e) {
        const files = e.target.files;
        const data = new FormData();
        data.append("file", files[0]);
        data.append("upload_preset", "farmersspecial");
    
        const res = await fetch(
          "https://api.cloudinary.com/v1_1/dzqeffkmp/image/upload",
          {
            method: "POST",
            body: data,
          }
        );
        const file = await res.json();
        setValues({ image: file.secure_url });
      };

    const handleSelect = (e) => {
        const roleSelection = e.target.value;
        setValues({ 
            ...values,
            role: roleSelection 
        });
    };

    function clearForm() {
        const emptyForm = Object.fromEntries(
            Object.entries(values).map(([key]) => [key, ''])
        );
        setValues(emptyForm);
    };

    function setInitialForm() {
        setValues(initialState);
    };

    return {
        values,
        setValues,
        handleChange,
        handleImageUpload,
        clearForm,
        setInitialForm,
        handleSelect,
    }
};