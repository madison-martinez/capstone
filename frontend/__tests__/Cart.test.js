import React from 'react';
import { screen, render, fakeItem } from '../src/utils/testUtils';
import Cart from '../src/components/Cart';
import { waitFor } from '@testing-library/react';

const item = fakeItem();

describe('< Cart />', () => {
    it('should display empty cart when no items in state', async () => {
        const { container } = render(
        <Cart />);

        expect(container).toMatchSnapshot();
    });

    it('should display fake item correctly when added into cart state', async () => {
        render(<Cart />,  { initialState: { cart: { cartProducts: [item]}}});
        await waitFor(() => expect(screen.getByTestId('order-button')).toBeEnabled());
        expect(screen.getByText(item.description)).toBeInTheDocument();
    })
})