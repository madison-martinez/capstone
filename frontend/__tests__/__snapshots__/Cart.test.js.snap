// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Cart /> should display empty cart when no items in state 1`] = `
<div>
  <h1>
    Your cart looks pretty empty.
  </h1>
  <h3>
    You have 
    0
     items in your cart!
  </h3>
  <div
    class="CartProduct__CartList-sc-17ufdy6-0 xzHqC"
  />
  <h2>
    Your total is: 
    $0.00
  </h2>
  <button
    data-testid="order-button"
    disabled=""
    type="button"
  >
    Place Order
  </button>
</div>
`;
